add-remove-list

Ibtisam34 opened the pull request:
- [ ] In this project,I had implement the CRUD (create, read, update, delete) methods. All the elements of the user interface will be fully functional and my application will be completed.

DammyShittu commented on src/models/all.js:
```diff
@@ -0,0 +1,54 @@
+const taskArr = [];
+
+// display task
+const displayTask = (task) => {
+  const listItem = `
+  <li>
+    <div class="check">
+      <input type="checkbox" name="checkbox" class="checkbox" id="${task.description}">
+      <input type="text" class="task-description" name="${task.description}" class="task-name" id="task-name" value="${task.description}">
+    </div>
+    <div class="actions">
+      <i class="fa-solid fa-pen-to-square edit"></i>
+      <i class="fa-solid fa-trash-can del"></i>
+    </div>
+  </li>`;
+  return listItem;
+};
```
- [ ] I am unable to delete a task because I cannot find the button. Kindly ensure the delete button is displayed and the functionality works as required. The index properties should be updated when a task is deleted.

  ![kfkfk](https://user-images.githubusercontent.com/54848363/201153805-24c442f5-2fa6-404e-b795-d1abb3906c9e.jpg)

Qoosim replied:
- [ ] As observed by the previous reviewer, there no button to delete todo on your page. Kindly look into it and add the button to delete

![your-page](https://user-images.githubusercontent.com/26597432/201193621-6e9d3bf0-1871-4418-94bf-7b1066dc91b6.png)

DammyShittu commented on src/index.js:
```diff
@@ -1,12 +1,54 @@
+
+  if (e.target.classList.contains('edit')) {
+    editTask(task);
+  }
```
- [ ] The edit functionality does not work. The edited task should also reflect on the local storage and when I reload the page the update description should be present, not the old one.

DammyShittu reviewed (changes requested):
Hi @Ibtisam34,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ibtisam34 commented:
Kindly  how did you check my work like how did you saw my form because
from my side I see only form without  list

On Thu, 10 Nov 2022, 7:38 pm Adedamola Shittu, ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi @Ibtisam34 <https://github.com/Ibtisam34>,
> Highlights
>
> ✔️ Descriptive PR
> ✔️ Passing linters
> ✔️ Good branch name
>
> Good job so far!
> There are some issues that you still need to work on to go to the next
> project but you are almost there!
> Required Changes ♻️
>
> *Check the comments under the review.*
> Optional suggestions
>
> *Every comment with the [OPTIONAL] prefix is not crucial enough to stop
> the approval of this PR. However, I strongly recommend you to take them
> into account as they can make your code better.*
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> *As described in the Code reviews limits policy
> <https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a
> limited number of reviews per project (check the exact number in your
> Dashboard). If you think that the code review was not fair, you can request
> a second opinion using this form <https://airtable.com/shrQAqnBwek5a0O0s>.*
> ------------------------------
>
> In src/models/all.js
> <https://github.com/Ibtisam34/To-do-list/pull/2#discussion_r1019371398>:
>
> > +const displayTask = (task) => {
>
> +  const listItem = `
>
> +  <li>
>
> +    <div class="check">
>
> +      <input type="checkbox" name="checkbox" class="checkbox" id="${task.description}">
>
> +      <input type="text" class="task-description" name="${task.description}" class="task-name" id="task-name" value="${task.description}">
>
> +    </div>
>
> +    <div class="actions">
>
> +      <i class="fa-solid fa-pen-to-square edit"></i>
>
> +      <i class="fa-solid fa-trash-can del"></i>
>
> +    </div>
>
> +  </li>`;
>
> +  return listItem;
>
> +};
>
>
>
>    -
>
>    I am unable to delete a task because I cannot find the button. Kindly
>    ensure the delete button is displayed and the functionality works as
>    required. The index properties should be updated when a task is deleted.
>
>    [image: kfkfk]
>    <https://user-images.githubusercontent.com/54848363/201153805-24c442f5-2fa6-404e-b795-d1abb3906c9e.jpg>
>
> ------------------------------
>
> In src/index.js
> <https://github.com/Ibtisam34/To-do-list/pull/2#discussion_r1019372742>:
>
> > +const editTask = (task) => {
>
> +  const tasks = JSON.parse(localStorage.getItem('tasks'));
>
> +  const taskItem = task.children[0].children[1].name;
>
> +  const taskIndex = tasks.findIndex((x) => x.description === taskItem);
>
> +  const taskName = task.querySelector('#task-name').value;
>
> +  tasks[taskIndex].description = taskName;
>
> +  localStorage.setItem('tasks', JSON.stringify(tasks));
>
> +};
>
>
>
> -UI.addBookToList(Todo);
>
> \ No newline at end of file
>
> +lists.addEventListener('click', (e) => {
>
> +  const task = e.target.parentElement.parentElement;
>
> +  if (e.target.classList.contains('del')) {
>
> +    deleteTask(task, e.target);
>
> +  }
>
> +
>
> +  if (e.target.classList.contains('edit')) {
>
> +    editTask(task);
>
> +  }
>
>
>
>    - The edit functionality does not work. The edited task should also
>    reflect on the local storage and when I reload the page the update
>    description should be present, not the old one.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ibtisam34/To-do-list/pull/2#pullrequestreview-1176134502>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZYCIBQZ7JZPTCJFHWFYAULWHUQJDANCNFSM6AAAAAAR4XQ73I>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Qoosim commented on src/index.js:
```diff
@@ -1,12 +1,54 @@
 
-// eslint-disable-next-line no-undef, no-unused-vars
-const book = new Todo(desc, completed, id);
+const editTask = (task) => {
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  const taskItem = task.children[0].children[1].name;
+  const taskIndex = tasks.findIndex((x) => x.description === taskItem);
+  const taskName = task.querySelector('#task-name').value;
+  tasks[taskIndex].description = taskName;
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
```
- [ ] As noticed by the previous reviewer, the edit function does not take effect both on your page and in the local storage. Kindly look into this and fix it.

Qoosim reviewed (changes requested):
#  *Status: Required Changes* :recycle: 

Hi @Ibtisam34, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights


-  No linter errors. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 

![](https://user-images.githubusercontent.com/33840476/192236577-e2d71173-44d8-4ac8-9615-bf78ba730792.gif)

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ibtisam34 commented:
@Qoosim  plz could you check again  the button  is there I put it before  I even pushed I made changes ,I added pin instead  of button to remove  the error could you check again those things I changed  them just couple  hours ago

leonardodiasb commented on src/index.js:
```diff
@@ -1,12 +1,55 @@
+    deleteTask(task, e.target);
+  }
+
+  if (e.target.classList.contains('edit')) {
+    editTask(task);
```
- [OPTIONAL] All the actions are working and the edit functionality too, but it would be great if the edit functionality happens right after the user makes a change to the input like adding or removing words so he/she doesn't need to click on the button to record the changes.

leonardodiasb reviewed (approved):
# Status: ✔️ Project Approved ✔️ 

Hello @ 🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

## Highlights

✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ibtisam34 committed:
added files

Ibtisam34 committed:
make changes
